### PR TITLE
Add documentation for ingress testing of Kong Proxy

### PR DESCRIPTION
**Pull request by ced0ps**
### What does this PR do?

This PR adds documentation on how to manually test the Ingress functionality of Kong Proxy
### What is needed from the reviewers?
- Grammar checking
- Review comprehensiveness 
### Why is this piece of content here?

This is documentation for the Connectivity Team but it could be used by customers as well.

### CODE REVIEW

**Comment by ced0ps** [comment]
image

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+title: "Manual test of Kong Ingress functionality"
+description: >
+  Documentation of how to test the ingress functionality of Kong Proxy.
+weight: 40
+---
+
+Tue, 23. Apr 2024
```
Comment (lyind) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:8]: Date isn't needed here, evident from Git metadata.
```suggestion
```

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Tue, 23. Apr 2024 
+
+### Step 1: Set Up AWS Infrastructure for Kong
+Ensure that the Kong App is installed in the Workload Cluster.
+Make sure that the Kong Proxy service is configured with an attached Amazon Web Services (AWS) Load Balancer (LB).
+```bash
+k get -n kong-app svc kong-app-kong-app-proxy -o jsonpath='{.status.loadBalancer.ingress[].hostname}'
+# Example Output
+# a62e0e6cd3ac3434c9c5f688277e56c8-1838414965.eu-central-1.elb.amazonaws.com
```
Comment (lyind) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:16]: Anonymize LOL
```suggestion
# c62e1e6bb3ac3535c9e5f698277e57c8-2828514061.eu-central-1.elb.amazonaws.com
```

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+### Step 2: Apply Configuration to Kong
+Apply a values.yaml file to your Kong installation.
```
Comment (lyind) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:20]:
```suggestion
Apply a `values.yaml` file to your Kong installation.

```

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  - host: hello.kong.<wc-cluster-base-domain>
+    paths:
+    - path: /
+      pathType: Prefix
+  tls:
+  - secretName: hello-world-tls
+    hosts:
+    - hello.kong.<wc-cluster-base-domain>
+```
+Step 4: Test the Kong Proxy
```
Comment (lyind) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:49]:
```suggestion
### Step 4: Test the Kong Proxy
```

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+After applying the ingress configuration, test the Kong proxy functionality by sending requests to the `hello-world-app`.
+Verify the routing of requests through the Kong proxy and check the responses to ensure they are being processed correctly.
+Additionally, you can test the SSL/TLS configuration by accessing the hello.kong. URL over HTTPS and ensuring that the hello-world-tls certificate is being served correctly.
```
Comment (lyind) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:52]:
```suggestion
Additionally, you can test the SSL/TLS configuration by accessing the `hello.kong.<wc-cluster-base-domain>` URL over HTTPS and ensuring that the `hello-world-tls` certificate is being served correctly.
```

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+Replace with your actual wildcard cluster base domain and with the specific image tag you are using for the Kong installation.
```
Comment (lyind) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:54]:
```suggestion
Replace `<wc-cluster-base-domain>` above with your actual wildcard cluster base domain and `<kong-image-tag>` with the specific image tag you are using for the Kong installation.
```

**Review by lyind — APPROVED** [review]
Nice work, please have a look at my minor suggestions.

**Comment by ubergesundheit** [comment]
Not sure I like the order of step 1 and 2. IMO it makes sense to install with the right values the first time, then checking if the AWS LB was created.

Maybe also add a "Step 0" where you list the required prerequisites like the cluster domain.

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+```
+- Workload Cluster base domain
+```bash
+kubectl get -n giantswarm configmap coredns-chart-values -o jsonpath='{.data.values}' | grep baseDomain
+# Example Output
+# baseDomain: qw54m.k8s.gaia.eu-central-1.aws.gigantic.io
+```
+
+### Step 1: Deploy Kong OSS
+
```
Comment (ced0ps) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:23]:
```suggestion

```

**Review by ubergesundheit — APPROVED** [review]
nice

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+### Step 0: Prerequisites
+- external-dns-app >3.1.0
+```bash
+helm list -n kube-system | grep external-dns
+# Example Output
+# external-dns   kube-system   4   2024-04-23 09:37:02 UTC deployed   external-dns-app-3.1.0
+```
+- Workload Cluster base domain
+```bash
+kubectl get -n giantswarm configmap coredns-chart-values -o jsonpath='{.data.values}' | grep baseDomain
```
[the remark this answers is not in the document]
Comment (ced0ps) [reply content/docs/product/connectivity/kong/kong-ingress-testing.md]: I added an alternative command for CAPI clusters.

File: content/docs/product/connectivity/kong/kong-ingress-testing.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+### Step 0: Prerequisites
+- external-dns-app >3.1.0
+```bash
+helm list -n kube-system | grep external-dns
+# Example Output
+# external-dns   kube-system   4   2024-04-23 09:37:02 UTC deployed   external-dns-app-3.1.0
+```
+- Workload Cluster base domain
+```bash
+# Legacy clusters
```
Comment (ced0ps) [inline content/docs/product/connectivity/kong/kong-ingress-testing.md:17]:
```suggestion
# Vintage clusters

```

**Pull request merged**